extensions/openshift-ai/src/openshiftai.ts

```diff
@@ -0,0 +1,299 @@
+    if (!raw) return [];
+    // split raw string by token separator
+    return raw.split(TOKEN_SEPARATOR).map(str => {
+      const [token, baseURL] = str.split(INFO_SEPARATOR);
+      return {
+        token,
+        baseURL,
+      };
+    });
+  }
+
+  /**
+   * Save connection info to secret storage
+   * @param token
+   * @param baseURL
+   * @private
+   */
+  private async saveConnectionInfo(token: string, baseURL: string): Promise<void> {
+    // get existing tokens
+    const tokens = (await this.getConnectionInfos()).map(t => `${t.token}|${t.baseURL}`);
+    // concat new token with existing tokens
+    const raw = [...tokens, `${token}${INFO_SEPARATOR}${baseURL}`].join(TOKEN_SEPARATOR);
+    // save to secret storage
+    await this.secrets.store(TOKENS_KEY, raw);
+  }
+
+  private async removeConnectionInfo(token: string, baseURL: string): Promise<void> {
+    // get existing tokens
+    const tokens = await this.getConnectionInfos();
+    // filter out the token
+    const raw = tokens.filter(t => t.token !== token || t.baseURL !== baseURL).join(TOKEN_SEPARATOR);
+    // save to secret storage
+    await this.secrets.store(TOKENS_KEY, raw);
+  }
+
+  protected async listModels(baseURL: string, token: string): Promise<Array<{ label: string }>> {
+    const res = await fetch(`${baseURL}/models`, {
+      headers: {
+        Authorization: `Bearer ${token}`,
+      },
+    });
+    if (res.status !== 200) throw new Error('failed to list models');
+    const body = await res.json();
+
+    if (!('data' in body)) throw new Error(`malformed response from ${baseURL}`);
+    if (!Array.isArray(body.data)) throw new Error(`malformed response from ${baseURL}: data is not an array`);
+
+    return body.data.map((model: { id: string }) => ({ label: model.id }));
+  }
+
+  private async getToken(coreAPI: CoreV1Api, namespace: string, runtime: string): Promise<string> {
+    const secrets = await coreAPI.listNamespacedSecret({ namespace });
+    for(const secret of secrets.items) {
+      if (secret.metadata?.annotations?.['kubernetes.io/service-account.name'] === runtime) {
+        if (secret.data?.['token']) {
+          return Buffer.from(secret.data['token'], 'base64').toString('utf-8');
+        }
+      }
+    }
+    throw new Error(`Failed to find token for runtime ${runtime}`);
+  }
+
+  private async getInferenceServices(url: string, token: string): Promise<ConnectionInfo[]> {
+    const urls: ConnectionInfo[] = [];
+    try {
+      const user = {
+        name: token,
+        token,
+      };
+      const cluster = {
+        name: 'openshift-ai',
+        server: url,
+        skipTLSVerify: true,
+      };
+      const context = {
+        cluster: cluster.name,
+        user: user.name,
+        name: 'openshift-ai',
+      };
+      const kc = new KubeConfig();
+      kc.loadFromOptions({
+        clusters: [cluster],
+        users: [user],
+        contexts: [context],
+        currentContext: context.name,
+      });
```

Comment:
I think it should use `kubernetes.getKubeconfig()` to get the kube config file configured in the tool

Reply:
I'm not using the Kube config file as the cluster may not be configured there but an in memory config file